src/sim/trace-extract/event_desc.proto

```diff
@@ -11,6 +11,8 @@ message EventDescriptor {
     REMOVE_MACHINE = 1;
     UPDATE_MACHINE = 2;
     TASK_END_RUNTIME = 3;
+  // this is a 'fake' event, used to indicate graph changes due to pinning, etc
+    TASK_ASSIGNMENT_CHANGED = 4;
```

Comment:
I think we should leave it at a more generic name, since this would also cover e.g. preemption and migration (which don't change the flow graph, but do require re-running).